src/rdo.rs

```diff
@@ -1189,42 +1219,11 @@ fn intra_frame_rdo_mode_decision<T: Pixel>(
     };
 
     satds.sort_by_key(|a| a.1);
-
-    // Find mode with lowest rate cost
-    let mut z = 32768;
-    let probs_all = if fi.frame_type.has_inter() {
-      cw.get_cdf_intra_mode(bsize)
-    } else {
-      cw.get_cdf_intra_mode_kf(tile_bo)
-    }
-    .iter()
-    .take(INTRA_MODES)
-    .map(|&a| {
-      let d = z - a;
-      z = a;
-      d
-    })
-    .collect::<ArrayVec<[_; INTRA_MODES]>>();
-
-    let mut probs = intra_mode_set
-      .iter()
-      .map(|&a| (a, probs_all[a as usize]))
-      .collect::<ArrayVec<[_; INTRA_MODES]>>();
-    probs.sort_by_key(|a| !a.1);
-
-    probs
-      .iter()
-      .take(num_modes_rdo / 2)
-      .for_each(|&(luma_mode, _prob)| modes.push(luma_mode));
     satds.iter().take(num_modes_rdo).for_each(|&(luma_mode, _stad)| {
       if !modes.contains(&luma_mode) {
         modes.push(luma_mode)
       }
     });
-  } else {
```

Comment:
@tdaede, Wanted to let you know the change that barrbrain is trying to use before-split tx_size to prune intra prediction modes when tx split occurred. And yes, this will reduce the encoding time with some loss of coding gain, especially with recently enabled tx-split for intra mode blocks in inter frames.